tiledb/sm/subarray/subarray.cc

```diff
@@ -1053,7 +1044,9 @@ void Subarray::set_layout(Layout layout) {
 }
 
 void Subarray::set_config(const QueryType query_type, const Config& config) {
-  config_.inherit(config);
+  merge_overlapping_ranges_ = config.get<bool>(
```

Comment:
This definetely improves performance but I am not sure about its maintainability. If the config class evolves we will need to update the constructor again. Plus, I am sure that there are many places in the code where this optimization can improve performance, so this opens the Pandora's box.